### PR TITLE
Rename Istio-Manager to Pilot

### PR DESCRIPTION
**Pull request by rshriram**
A site wide renaming of Istio-Manager to pilot in all docs, diagrams, slides, etc.

TODO:
1. Istioctl has CLI options (--managerAPIAddress) whose default value is "istio-manager:..."
   (fyi @ayj @esnible)
2. Installing Istio asks users to look for a service named istio-manager
  (fyi @andraxylia @frankbu )

We cannot change the above two references until we change the install scripts in istio/istio and change istioctl CLI as well. Once we fix those two components in the next 0.1.6 release, we can update these two references as well.

Since this is a large patch, I suggest reviewing the individual commits

### CODE REVIEW

**Comment by rshriram** [comment]
Also ran the link checker. It will fail because two links (pilot.html and pilot.md) won't exist until PR is merged.

```
- ./_site/docs/concepts/network-and-auth/auth.html
  *  External link https://istio.io/docs/concepts/traffic-management/pilot.html failed: 404 No error
- ./_site/docs/concepts/traffic-management/pilot.html
  *  External link https://github.com/istio/istio.github.io/edit/master/_docs/concepts/traffic-management/pilot.md failed: 404 No error
```

File: _docs/concepts/network-and-auth/auth.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ Istio Auth workflow consists of two phases, deployment and runtime. This section
 
 1.  When a pod is created, API Server mounts the key and certificate pair according to the service account using [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
 
-1.  [Istio Manager](https://istio.io/docs/concepts/traffic-management/manager.html) generates the config with proper key and certificate and secure naming information, 
+1.  [Istio Manager](https://istio.io/docs/concepts/traffic-management/pilot.html) generates the config with proper key and certificate and secure naming information,
```
Comment (frankbu) [inline _docs/concepts/network-and-auth/auth.md]: s/[Istio Manager]/[Pilot]/

**Comment by kyessenov** [comment]
@rshriram the option has already been renamed to `configAPIService`. The config gen is not automated, so @ayj do you mind regenerating the docs from the CLI tool?

**Comment by ayj** [comment]
I can refresh CLI reference docs in a follow-up PR

**Comment by ldemailly** [comment]
first link should be relative, second one I wonder why it's showing like this - does this happen on any new page (seems like a bug, I guess the edit link should be aware of the branch/pr)

File: _docs/concepts/network-and-auth/auth.md [hunk 2]
```diff
@@ -91,7 +91,7 @@ Istio Auth workflow consists of two phases, deployment and runtime. This section
 
 1.  When a pod is created, API Server mounts the key and certificate pair according to the service account using [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
 
-1.  [Istio Manager](https://istio.io/docs/concepts/traffic-management/manager.html) generates the config with proper key and certificate and secure naming information, 
+1.  [Pilot](https://istio.io/docs/concepts/traffic-management/pilot.html) generates the config with proper key and certificate and secure naming information,
```
Comment (ldemailly) [inline _docs/concepts/network-and-auth/auth.md]: please fix this link to use {{home}} or just ../traffic-management/pilot.html

not sure why someone hardcoded a bunch of https://istio.io/docs/ will file a seperate issue

**Comment by ldemailly** [comment]
should we wait until we have a 0.1.6 before changing the docs ?
if we push this PR without having an alpha archive, things will start breaking for users following instruction about `--configAPIservice` for instance

**Comment by rshriram** [comment]
Yes absolutely. Once 0.1.6 is out, we push this PR in.

**Comment by ldemailly** [comment]
we have a 0.1.6 now (to be confirmed as solid maybe before pushing this)

**Pull request merged**